qfieldsync/gui/package_dialog.py

```diff
@@ -150,24 +159,40 @@ def get_export_filename_suggestion(self) -> str:
         )
         return str(full_project_name_suggestion)
 
-    def show_package_page(self):
-        self.nextButton.setVisible(False)
-        self.button_box.setVisible(True)
-        self.stackedWidget.setCurrentWidget(self.packagePage)
-
-    def run_package_project(self) -> None:
+    def check_path_length_and_extension(self):
+        """Check if the file path exceeds 200 characters and the extension is .qgs."""
         export_packaged_project = Path(self.packagedProjectFileWidget.filePath())
 
-        if export_packaged_project.suffix != ".qgs":
-            QMessageBox.critical(
-                self,
-                self.tr("Invalid Filename"),
-                self.tr('The filename must have a ".qgs" extension.'),
+        if export_packaged_project.suffix.lower() != ".qgs":
+            self._set_feedback(
+                self.tr('The filename must have a ".qgs" extension'), disable_save=True
             )
             return
 
-        else:
-            self.package_project()
+        if len(export_packaged_project.as_posix()) > self.MAX_LENGTH_CHARS_FILEPATH:
+            self._set_feedback(
+                self.tr(
+                    "Warning: File path exceeds 200 characters. "
+                    "Longer paths may not be handled properly by your file system."
+                ),
+                disable_save=True,
+            )
+            return
+
+        self._set_feedback("", disable_save=False)
```

Comment:
Can you avoid `_set_packaged_project_filename_feedback` function and show the proper feedback in the end of the current function? I know it is not following "separation of concerns" logic, but in this case it is following "simplification" of the class API? Also it will allow you to show multiple errors at once if the validation goes further.